tests/Integration/CreateUserTrait.php

```diff
@@ -20,7 +23,45 @@
  */
 trait CreateUserTrait
 {
-    protected function createUser(?string $password = null): InternalUser
+    protected function createExternalUser(?string $preferredUsername = null, string $actorType = 'Person'): ExternalUser
+    {
+        /** @var EntityManagerInterface $entityManager */
+        $entityManager = $this->getContainer()->get(EntityManagerInterface::class);
+
+        if (null === $preferredUsername) {
+            $preferredUsername = sprintf('bob.%s', uniqid());
+        }
+
+        $externalUserId = sprintf('https://example.com/user/%s', $preferredUsername);
+
+        $externalUser = new ExternalUser(
+            Uuid::uuid4()->toString(),
+            $externalUserId,
+            hash('sha256', $externalUserId),
+            $preferredUsername,
+            sprintf('https://example.com/user/%s/inbox', $preferredUsername),
+            sprintf('https://example.com/user/%s/outbox', $preferredUsername)
+        );
+
+        if ('Person' === $actorType) {
```

Comment:
What about the other three actors(Application, Group and Service)?

Reply:
@fgervasi not supported yet. Created an issue for this: https://github.com/mitra-social/mitra-backend/issues/35